### DISCUSSION TITLE
Why are we forced to use new method for getting confirmed transaction?

### DISCUSSION THREAD

**Discussion by nukec** [bmresearch/Solnet#214] [General]
Stable version of mainnet-beta is 1.6.26.

With solnet 0.4.8 I am forced to use following method for getting confirmed transaction:
`GetTransactionAsync`

```
            RequestResult<string>? transactionResult = await this._rpcClient.SendTransactionAsync(transaction: tx, commitment: Commitment.Confirmed);
            await this._rpcClient.GetTransactionAsync(transactionResult.Result, Commitment.Confirmed);
```

Because mainnet-beta is 1.6.26 this doesn't work as it gives `method not found` rpc error response. I also cannot use obsolete method `getConfirmedTransactionAsync` or however the name was which uses 1.6.26 method, due to my code analyzer tool. Problem is obv solved if I could just ignore and use obsolete method but the thing is that in my opinion method is not obsolete if it is required for 1.6.26. Obsolete should only be added once 1.7+ is stable, which uses new method.

Thoughts?

**Comment by tiago18c** [comment]
Solana documentation declared these methods as deprecated, so we followed them. It's also a good reminder that very soon they will become unsupported so you have to be prepared.

To overcome the code analyzer issue,  just disable this specific warning for that line.

```
#pragma warning disable CS0618 // Type or member is obsolete
            _rpcClient.GetConfirmedTransaction("tx hash");
#pragma warning restore CS0618 // Type or member is obsolete
```
